pertpy/tools/_metadata.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import anndata
+import pandas as pd
+from scanpy import settings
+
+from pertpy.data._dataloader import _download
+
+
+class MetaData:
+    """Utilities to fetch cell or perturbation metadata."""
+
+    def __init__(self):
+        """Download cell line metadata from DepMap"""
+        cell_line_file_name = "sample_info.csv"
+        cell_line_file_path = settings.cachedir.__str__() + "/" + cell_line_file_name
+        if not Path(cell_line_file_path).exists():
+            print("No DepMap metadata file found. Start downloading now.")
+            _download(
+                url="https://ndownloader.figshare.com/files/35020903",
+                output_file_name=cell_line_file_name,
+                output_path=settings.cachedir,
+                is_zip=False,
+            )
+            self.cell_line_meta = pd.read_csv(settings.cachedir.__str__() + "/" + cell_line_file_name)
+        else:
+            self.cell_line_meta = pd.read_csv(cell_line_file_path)
+
+    def annotate_cell_lines(
+        self,
+        adata: anndata,
+        cell_line_column: str = "DepMap_ID",
+        cell_line_type: str = "DepMap_ID",
+        cell_line_information: list[str] = None,
+        copy: bool = False,
+    ) -> anndata:
+        """Fetch cell line annotation.
+
+        For each cell, we fetch cell line annotation from Dependency Map (DepMap).
```

Comment:
This is not quite in consideration at the moment, but we'll probably have sources other than DepMap for cell line annotations in the future - just a thought.

Reply:
Here's an example of another source of cell line information - ideally you'd want another internal function that aligns this with DepMap when loading in the csv, for instance. https://www.cancerrxgene.org/celllines